### PR TITLE
[HOTFIX] Validate note name

### PR DESCRIPTION
**Pull request by jongyoul**
### What is this PR for?
Checking invalid note name like './';

### What type of PR is it?
Hot Fix

### Todos
* [x] - Add validation logic for note names

### What is the Jira issue?
* N/A

### How should this be tested?
* Invalid note name should be rejected

### Screenshots (if appropriate)
* N/A

### Questions:
* Does the license files need to update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: zeppelin-server/src/test/java/org/apache/zeppelin/service/NotebookServiceTest.java [hunk 1]
```diff
@@ -528,5 +528,17 @@ void testNormalizeNotePath() throws IOException {
...
     }
+    try {
+      notebookService.normalizeNotePath("%2e%2e/%2e%2e/tmp/test222");
+      fail("Should fail");
+    } catch (IOException e) {
+      assertEquals("Note name can not contain '..'", e.getMessage());
+    }
+    try {
+      notebookService.normalizeNotePath("./");
+      fail("shold fail");
```
Comment (Reamer) [inline zeppelin-server/src/test/java/org/apache/zeppelin/service/NotebookServiceTest.java:539]: small typo

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java [hunk 2]
```diff
@@ -236,6 +237,12 @@ String normalizeNotePath(String notePath) throws IOException {
     }
 
     notePath = notePath.replace("\r", " ").replace("\n", " ");
+
+    notePath = URLDecoder.decode(notePath, "UTF-8");
```
Comment (Reamer) [inline zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java:241]: It's better to use `URLDecoder.decode(value, StandardCharsets.UTF_8.toString());`
Unfortunately, I have not found a method where you can give a charset.

Comment (Reamer) [reply zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java:241]:
```suggestion
    notePath = URLDecoder.decode(notePath, StandardCharsets.UTF_8.toString());
```

File: zeppelin-server/src/test/java/org/apache/zeppelin/service/NotebookServiceTest.java (hunk 1 above)
Comment (Reamer) [reply zeppelin-server/src/test/java/org/apache/zeppelin/service/NotebookServiceTest.java:539]:
```suggestion
      fail("Should fail");
```

**Review by Reamer — DISMISSED** [review]

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java (hunk 2 above)
Comment (huage1994) [reply zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java]: CI prints
```Error:  /home/runner/work/zeppelin/zeppelin/zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java:[241,60] package StandardCharsets does not exist```

The StandardCharsets import statement  is missing.
Maybe ```import java.nio.charset.StandardCharsets;```   is ok

**Review by huage1994 — APPROVED** [review]

**Pull request merged**